app/src/main/kotlin/co/touchlab/droidconandroid/VoteFragment.kt

```diff
@@ -63,87 +64,129 @@ class ScheduleFragment : Fragment(), FilterableFragmentInterface
         EventBusExt.getDefault().unregister(this)
     }
 
-    public fun onEventMainThread(eventDetailTask: RefreshScheduleDataKot)
-    {
-        Handler().post(RefreshRunnable())
-    }
-
     override fun onActivityCreated(savedInstanceState: Bundle?) {
         super<Fragment>.onActivityCreated(savedInstanceState)
 
-        Handler().post(RefreshRunnable())
+        //empty view
+        empty = view.findViewById(R.id.empty_list) as TextView
+
+        //swipe to refresh
+        swipeContainer = view.findViewById(R.id.swipeContainer) as SwipeRefreshLayout
+        initSwipeToRefresh()
+
+        //recycler list
+        rv = view.findViewById(R.id.rv) as RecyclerView
+        rv!!.layoutManager = LinearLayoutManager(activity)
+
+        (activity as AppCompatActivity).supportActionBar.setTitle(R.string.vote)
+        TaskQueue.loadQueueDefault(activity).execute(GetDbTalkSubmissionTask(true))
     }
 
-    inner class RefreshRunnable(): Runnable
-    {
-        override fun run() {
-            val pager = getView().findViewById(R.id.pager)!! as ViewPager
-            val tabs = getView().findViewById(R.id.tabs)!! as TabLayout
-            val tabWrapper = getView().findViewById(R.id.tabs_wrapper)
-            if (getArguments().getBoolean(ALL_EVENTS)) {
-                tabWrapper.setBackgroundColor(getResources().getColor(R.color.primary))
-            } else {
-                tabWrapper.setBackgroundColor(getResources().getColor(R.color.blue_grey))
-            }
+    override fun onCreateOptionsMenu(menu: Menu?, inflater: MenuInflater?) {
+        inflater!!.inflate(R.menu.vote_list, menu)
```

Comment:
A new voted action is being added to the action bar every time the app is rotated. Perhaps this needs to  check if the menu is already there, or delegate it to the activity?